### DISCUSSION TITLE
Why is `ScaleToBounds` scaling to an interval multiplied by 0.95?

### DISCUSSION THREAD

**Discussion by andreimargeloiu** [cornellius-gp/gpytorch#1740] [Q&A] [answered]
Looking at the [source code](https://docs.gpytorch.ai/en/stable/_modules/gpytorch/utils/grid.html#ScaleToBounds) for ScaleToBounds, I noticed that it scales to an interval multiplied by 0.95. For example `ScaleToBounds(1, 2)` will output numbers between 0.95 and 1.9. 

What's the rationale behind the decision to multiply by 0.95?

**Comment by andreimargeloiu** [comment, marked as answer]
Any thoughts?

**Reply by wjmaddox** [reply]
I think to prevent edge case errors of hitting exactly [0.95].

**Reply by jacobrgardner** [reply]
I believe the reason we do this is because SKI ends up with some pretty weird boundary conditions whenever you have data that's really close to the boundary, because you end up with fewer inducing points than expected on either side. Our goal in scaling is to always avoid `left_boundary_pts` from every actually having data here: https://github.com/cornellius-gp/gpytorch/blob/f06004e888542289d19c155e4e2ad35aad0573b7/gpytorch/utils/interpolation.py#L117-L120

because the "fix" for the boundary condition involved ends up having pretty low fidelity. Similarly we don't want the right side condition to ever trigger either:
https://github.com/cornellius-gp/gpytorch/blob/f06004e888542289d19c155e4e2ad35aad0573b7/gpytorch/utils/interpolation.py#L135-L138

**Reply by gpleiss** [reply]
To build upon what @jacobrgardner mentioned here - SKI is a lot faster if we can avoid triggering the edge cases (or at least this was the case when we first implemented it!).
